server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsNodeResponse.java

```diff
@@ -81,6 +88,24 @@ public ClusterStatsNodeResponse(
         this.clusterStatus = clusterStatus;
     }
 
+    public ClusterStatsNodeResponse(
+        DiscoveryNode node,
+        @Nullable ClusterHealthStatus clusterStatus,
+        NodeInfo nodeInfo,
+        NodeStats nodeStats,
+        ShardStats[] shardsStats,
+        boolean useAggregatedNodeLevelResponses
+    ) {
+        super(node);
+        this.nodeInfo = nodeInfo;
+        this.nodeStats = nodeStats;
+        if (useAggregatedNodeLevelResponses) {
```

Comment:
We only create `aggregatedNodeLevelStats` when the param is passed. Else we do not. This reduces the request size when not required.

Reply:
I am still unclear why you need useAggregatedNodeLevelResponses boolean in response as well. If recieved it in request, you will send the response accordingly.

Reply:
We do not use the flag in the response. It is just used for generating the response and is not passed around.